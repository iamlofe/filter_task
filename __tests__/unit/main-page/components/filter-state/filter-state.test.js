import React from 'react';
import { shallow, mount } from 'enzyme';
import { List } from 'immutable';

import FilterState from 'main-page/components/filter-state/filter-state';
import { getMockListItems } from '../../mock-store/mock-filter-reducer';

describe('component filter state', () => {
    const baseProps = {
        listItems: getMockListItems(),
        selectedItems: new List([3, 4]),
        label: 'pow'
    };

    it('should make snapshot without crashing', () => {
        const filteState = shallow(<FilterState {...baseProps} />);
        expect(filteState).toMatchSnapshot();
    });

    it("should don't render element if selectedItems is empty", () => {
        const filteState = shallow(<FilterState {...baseProps} selectedItems={new List()} />);
        expect(filteState).toMatchSnapshot();
    });

    it('should show title', () => {
        const filteState = shallow(<FilterState {...baseProps} />);
        expect(filteState.find('.filter-container__display-context-title').text()).toEqual(`${baseProps.label}:`);
    });

    it('should have truth length', () => {
        const filteState = mount(<FilterState {...baseProps} />);
        expect(filteState.find('.filter-container__display-context-list').children().length).toEqual(baseProps.selectedItems.count());
    });
});
