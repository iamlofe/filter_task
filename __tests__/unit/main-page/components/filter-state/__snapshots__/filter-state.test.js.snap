// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component filter state should don't render element if selectedItems is empty 1`] = `<Fragment />`;

exports[`component filter state should make snapshot without crashing 1`] = `
<Fragment>
  <div
    className="filter-container__display-context"
  >
    <div
      className="filter-container__display-context-title"
    >
      pow
      :
    </div>
    <div
      className="filter-container__display-context-list"
    >
      <div
        key="3"
      >
        results
      </div>
      <div
        key="4"
      >
        results2
      </div>
    </div>
  </div>
</Fragment>
`;
